log4j-core-test/src/test/java/org/apache/logging/log4j/core/util/ShutdownCallbackRegistryTest.java

```diff
@@ -28,20 +28,16 @@
 import org.apache.logging.log4j.core.impl.Log4jContextFactory;
 import org.apache.logging.log4j.core.selector.ContextSelector;
 import org.apache.logging.log4j.core.test.junit.LoggerContextSource;
+import org.apache.logging.log4j.core.test.junit.TestBinding;
 import org.apache.logging.log4j.plugins.Singleton;
-import org.apache.logging.log4j.plugins.SingletonFactory;
 import org.apache.logging.log4j.status.StatusLogger;
 import org.junit.jupiter.api.Test;
 
+@TestBinding(api = ShutdownCallbackRegistry.class, implementation = ShutdownCallbackRegistryTest.Registry.class)
```

Comment:
Since all test bindings are registered as initial bindings in the new DSL, that helps avoid the need for the old `bootstrap` boolean flag in `@LoggerContextSource`.